zebedee-cms/src/main/java/com/github/onsdigital/zebedee/model/approval/tasks/CollectionPdfGenerator.java

```diff
@@ -67,87 +48,60 @@ public CollectionPdfGenerator(PdfService pdfService) {
 
     /**
      * @param collection
-     * @param collectionWriter
+     * @param contentReader
+     * @param contentWriter
      * @param collectionContent
      * @throws ZebedeeException
      */
-    public void generatePDFsForCollection(Collection collection, CollectionWriter collectionWriter,
+    public void generatePDFsForCollection(Collection collection, ContentReader contentReader, ContentWriter contentWriter,
                                           List<ContentDetail> collectionContent) throws ZebedeeException {
         List<ContentDetail> filtered = filterPDFContent(collectionContent);
 
         int index = 1;
         for (ContentDetail detail : filtered) {
-            generatePDFForContent(collection, collectionWriter.getReviewed(), detail.uri);
+            generatePDFForContent(collection, contentWriter, detail.uri, detail.description.language);
+
+            if (!ContentLanguage.WELSH.getId().equals(detail.description.language)) {
+                // If this is an English version, check if a Welsh data file exists
+
+                Resource r;
+                try {
+                    r = contentReader.getResource(detail.uri + "/" + ContentLanguage.WELSH.getDataFileName());
+                } catch (Exception e) {
+                    // It's ok. If there is no Welsh data file, we won't generate a Welsh PDF
+                    r = null;
+                }
+
+                if (r != null) {
+                    generatePDFForContent(collection, contentWriter, detail.uri, ContentLanguage.WELSH.getId());
+                }
+            }
```

Comment:
Did you test it with a collection containing only Welsh changes? 
The problem is when the collection contains both English and Welsh changes. In that case we only receive one `ContentDetail` object here for that url, with the English info. This is because we use a `Set` [while finding them](https://github.com/ONSdigital/zebedee/blob/b72fad73eeaeee792d22effc05fca874c4891ff6/zebedee-cms/src/main/java/com/github/onsdigital/zebedee/util/ContentDetailUtil.java#L31) and the [equals method](https://github.com/ONSdigital/zebedee/blob/b72fad73eeaeee792d22effc05fca874c4891ff6/zebedee-cms/src/main/java/com/github/onsdigital/zebedee/json/ContentDetail.java#L144) for `ContentDetail` only takes into account the uri, not the language. I considered changing that but this is done deliberately and would break other functionality. 
If the Welsh version hasn't changed, I don't think we will regenerate the PDF as the check in line 69 would fail because there won't be a `data_cy.json` in the collection?

Reply:
OK, so the 'correct' way to fix it would be to fix the issue with the sets (if content has a different language then by definition it can't be 'equal' to the other language content). However, as you say, this will have a lot of implications due to zebedee's poor language handling (I can see the content tree functionality as one example that relies on this 'bug'). This does need fixing at some point (and I believe will actually have benefits with respect to other welsh language bugs) but is probably too big in scope for this change. Can I suggest instead we add a `FIXME` to this block explaining why it's needed with references to the `ContentDetailUtil` and the `ContentDetail::equals(…)`/`hashCode()` methods as you've detailed above. It's OK to do this as a workaround but we really need to clarify that that is what it is.